cognite/neat/core/extractors/rules_to_graph_capturing_sheet.py

```diff
@@ -0,0 +1,153 @@
+        "indent": kwargs.get("indent", 1),
+        "font_size": kwargs.get("font_size", 16),
+        "font_name": kwargs.get("font_name", "Helvetica"),
+    }
+
+    identifier_header_cfg = header_cfg.copy()
+    identifier_header_cfg["bg_color"] = "#FFB202"
+
+    return header_cfg, identifier_header_cfg
+
+
+def _add_uuid_identifiers(sheets, sheet_name, no_rows):
+    """Adds UUID-based auto identifier to a sheet"""
+    for i in range(no_rows):
+        fixed_name = to_graphql_name(sheet_name, "class", True)
+        sheets[sheet_name].write_formula(f"A{i+2}", f'=IF(ISBLANK(B{i+2}), "","{fixed_name}-{uuid.uuid4()}")')
+
+
+def _add_index_identifiers(sheets, sheet_name, no_rows):
+    """Adds index-based auto identifier to a sheet"""
+    for i in range(no_rows):
+        fixed_name = to_graphql_name(sheet_name, "class", True)
+        sheets[sheet_name].write_formula(f"A{i+2}", f'=IF(ISBLANK(B{i+2}), "","{fixed_name}-{i+1}")')
+
+
+def _set_column_width(sheet, no_columns, width):
+    """Sets the width of all columns in a sheet"""
+    for i in range(no_columns + 1):
+        sheet.set_column(i, i, width)
+
+
+def _add_drop_down_list(sheets, sheet_name, column, no_rows, value_sheet, value_column):
+    """Adds a drop down list to a column"""
+    logging.info(f"Adding drop down list to <{sheet_name}!{column}> with values from <{value_sheet}!{value_column}>")
+    for i in range(no_rows):
+        sheets[sheet_name].data_validation(
+            f"{column}{i+2}",
+            {
+                "validate": "list",
+                "source": f"={value_sheet}!{value_column}2:{value_column}{no_rows}",
+            },
+        )
+
+
+def rules2graph_capturing_sheet(
+    file_path: Path,
+    transformation_rules: TransformationRules,
+    no_rows: int = 1000,
+    use_index_id: bool = True,
+    use_uuid_id: bool = False,
+    add_drop_down_list: bool = True,
+    **kwargs,
+):
+    """Converts a TransformationRules object to a graph capturing sheet
+
+    Parameters
+    ----------
+    file_path : Path
+        File path to save the sheet to
+    transformation_rules : TransformationRules
+        The TransformationRules object to convert to the graph capturing sheet
+    use_uuid_id : bool, optional
+        Use UUID-based for automatic identifiers, by default False
+    use_index_id : bool, optional
+        Use Index-based for automatic identifiers, by default True
+    no_rows : int, optional
+        Number of rows for processing, by default 1000
+    add_drop_down_list : bool, optional
+        Add drop down selection for columns that contain linking properties, by default True
+    **kwargs : dict
+        Additional arguments to pass to the function dealing with header style
+
+    Notes
+    -----
+    no_rows should be set to maximum expected number of instances of any of the classes.
+    By default it is set to 1000, increase it accordingly if you have more instances.
+
+    """
+
+    workbook = xlsxwriter.Workbook(file_path)
```

Comment:
Can you use a context manager instead?
`with xlsxwriter.Workbook(file_path) as workbook` then you do not need the `workbook.close()` at the end. In addition, and more importantly, if an exception is raised the workbook is still closed. Now, you risk that the excel file becomes impossible to open as the program does not release its write lock on the file.